solutions/kubernetes/cluster-setup.sh

```diff
@@ -51,39 +48,41 @@ function setup_cluster {
     # Prepare nodes:
     pdsh -w ^/var/tmp/pdsh-hosts '/var/tmp/cluster-functions.sh --prepare'
 
-    echo "---------------------------------------[ Preparing Primary Node $PRIMARY_NODE ]--------------------------------------"
-    ssh -o 'StrictHostKeyChecking=no' "$PRIMARY_NODE" "/var/tmp/cluster-functions.sh --primary ${UNTAINT}"
+    add_secondary_separator "Setup Primary Node $PRIMARY_NODE"
+    ssh_primary_node "/var/tmp/cluster-functions.sh --primary ${UNTAINT}"
     check_status
-    sleep 10 #To be replaced with status check
-    JOIN_COMMAND=$(ssh -o 'StrictHostKeyChecking=no' "$PRIMARY_NODE" 'kubeadm token create --print-join-command --description "Token to join worker nodes"')
+    sleep 10 # To be replaced with status check
+    JOIN_COMMAND=$(ssh_primary_node 'kubeadm token create --print-join-command --description "Token to join worker nodes"')
     check_status "Failed fetch cluster join command"
 
-    echo $WORKER_NODES > /var/tmp/pdsh-hosts
-    # Join worker nodes.
-    JOIN_WORKER="/var/tmp/cluster-functions.sh --join-worker-nodes $JOIN_COMMAND"
-    pdsh -w ^/var/tmp/pdsh-hosts "$JOIN_WORKER"
-
-    # Label worker nodes.
-    for worker_node in $WORKER_NODES
-    do
-        ssh -o 'StrictHostKeyChecking=no' "$PRIMARY_NODE" "kubectl label node $worker_node" node-role.kubernetes.io/worker=worker
-        check_status "Failed to lable $worker_node"
-    done
+    if [ "$SINGLE_NODE_DEPLOYMENT" == "False" ]; then
```

Comment:
where `SINGLE_NODE_DEPLOYMENT` is defined?

Reply:
in functions.sh where its used to print deployment type.